webapp/publisher/snaps/build_views.py

```diff
@@ -530,21 +530,53 @@ def post_update_gh_webhooks(snap_name):
         return _handle_error(api_error)
 
     lp_snap = launchpad.get_snap_by_store_name(details["snap_name"])
+
+    if not lp_snap:
+        flask.flash(
+            "This snap is not linked with a GitHub repository", "negative"
+        )
+
+        return flask.redirect(
+            flask.url_for(".get_settings", snap_name=snap_name)
+        )
+
+    github = GitHub(flask.session.get("github_auth_secret"))
+
+    try:
+        github.get_user()
+    except Unauthorized:
+        return flask.redirect(f"/github/auth?back={flask.request.path}")
+
     gh_link = lp_snap["git_repository_url"][19:]
     gh_owner, gh_repo = gh_link.split("/")
 
-    github = GitHub(flask.session.get("github_auth_secret"))
+    # Remove old BSI webhook if present
     old_url = f"https://build.snapcraft.io/{gh_owner}/{gh_repo}/webhook/notify"
     old_hook = github.get_hook_by_url(gh_owner, gh_repo, old_url)
 
     if old_hook:
-        github.update_hook_url(
+        github.remove_hook(
             gh_owner,
             gh_repo,
             old_hook["id"],
-            f"https://snapcraft.io/{snap_name}/webhook/notify",
         )
 
-    return flask.redirect(
-        flask.url_for(".get_snap_builds", snap_name=snap_name)
+    # Remove current hook
+    github_hook_url = (
+        f"https://snapcraft.io/{gh_owner}/{gh_repo}/webhook/notify"
```

Comment:
this will only work for production environment and not staging.

Reply:
Yes, we have been setting up the production webhook always. It makes sense I will update it